packages/elements/src/components/ino-carousel/ino-carousel.tsx

```diff
@@ -0,0 +1,168 @@
+   */
+  @Prop() value?: any;
+
+  @Watch('value')
+  valueChanged(newVal: any) {
+    this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the current slide
+    this.selectSlide(newVal);
+    this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the new slide
+  }
+
+  /**
+   * Enables autoplay which causes slides to be changed automatically
+   */
+  @Prop() inoAutoplay: boolean = false;
+  @Watch('inoAutoplay')
+  inoAutoplayChanged() {
+    this.configureAutoplay();
+  }
+
+  /**
+   * Enables the slide animation
+   */
+  @Prop() inoAnimated: boolean = false;
+
+  /**
+   * Hides the arrow buttons
+   */
+  @Prop() inoHideButtons: boolean = false;
+
+  /**
+   * Restarts playback from the first slide upon reaching the last slide
+   */
+  @Prop() inoInfinite: boolean = false;
+
+  /**
+   * Sets the intermission between two slides (Unit: ms)
+   */
+  @Prop() inoIntermission: number = 5000;
+
+  /**
+   * Enables reverse playback of the slides
+   */
+  @Prop() inoReverse: boolean = false;
+
+  componentDidLoad(): void {
+    this.slides = this.getSlides();
+    this.configureSlides();
+    this.configureAutoplay();
+  }
+
+  /**
+   * adds a slide animation to the slide with the given index
+   * required to prevent the animation from playing when loading the component
+   * @param slide carousel slide
+   */
+  private addAnimationToSlide = (slide: HTMLInoCarouselSlideElement) => {
+    if (!this.inoAnimated) return;
+    if (!slide.classList.contains('ino-carousel--animated')) {
+      slide.classList.add('ino-carousel--animated');
+    }
+  };
+
+  private configureSlides = () => {
+    if (this.slides.length < 1) return;
+    if (!this.selectSlide(this.value)) this.slides[this.currentSlide].classList.add('ino-carousel-slide--selected');
+  };
+
+  private selectSlide(value: number): boolean {
+    let slideSelected = false;
+    this.slides.forEach((slide) => {
+      if (value === slide.value) {
+        slide.classList.add('ino-carousel-slide--selected');
+        this.currentSlide = this.slides.indexOf(slide);
+        slideSelected = true;
+      } else {
+        slide.classList.remove('ino-carousel-slide--selected');
+      }
+    });
+    return slideSelected;
+  }
+
+  private configureAutoplay = () => {
+    if (this.slides.length < 1) return;
+    if (this.inoAutoplay) {
+      this.timer = setInterval(this.nextSlide, this.inoIntermission);
+      return;
+    }
+    clearInterval(this.timer);
+  };
+
+  /**
+   * Switches to the next slide
+   * Required for autoplay to work
+   */
+  private nextSlide = () => {
+    if (this.slides.length < 1) throw new Error('There are no slides to display');
+    this.slides[this.currentSlide].classList.remove('ino-carousel-slide--selected');
+    this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to current slide
+    this.getNextSlide();
```

Comment:
for better readability, the `getNextSlide()` funtion should just return the next slide and not set it to the current. Otherwise, the name of the function is kinda misleading. 
Better would be:
```tsx
this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to current slide
this.currentSlide = this.getNextSlide();
this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the new slide
```

Reply:
You're right - I'll change it.